portal/tests/test_captcha_forms.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Code for Life
+#
+# Copyright (C) 2017, Ocado Innovation Limited
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# ADDITIONAL TERMS – Section 7 GNU General Public Licence
+#
+# This licence does not grant any right, title or interest in any “Ocado” logos,
+# trade names or the trademark “Ocado” or any other trademarks or domain names
+# owned by Ocado Innovation Limited or the Ocado group of companies or any other
+# distinctive brand features of “Ocado” as may be secured from time to time. You
+# must not distribute any modification of this program using the trademark
+# “Ocado” or claim any affiliation or association with Ocado or its employees.
+#
+# You are not authorised to use the name Ocado (or any of its trade names) or
+# the names of any author or contributor in advertising or for publicity purposes
+# pertaining to the distribution of this program, without the prior written
+# authorisation of Ocado.
+#
+# Any propagation, distribution or conveyance of this program must include this
+# copyright notice and these terms. You must not misrepresent the origins of this
+# program; modified versions of the program must be marked as such and not
+# identified as the original program.
+
+from django.test import TestCase
+from portal.helpers.captcha import remove_captcha_from_forms, is_captcha_in_form
+from django import forms
+from captcha.fields import ReCaptchaField
+
+
+class FormCaptchaTest(TestCase):
```

Comment:
I'm also wondering if we couldn't add a test where the recaptcha is here, try to access the page and assert it's failing?

Reply:
currently, the tests have a global flag that sets the recaptcha to false so it's not in the form, Google has provided a way to use selenium with reCaptcha V2 but it would require more refactoring.

Reply:
interesting - could that be like 2 pair programming lunch break c4l sessions? 🤔

Reply:
that's a really good idea! :) a good volunteer task

Reply:
made a task for it: https://github.com/ocadotechnology/codeforlife-portal/issues/668